### PR TITLE
Remaining Bundle changes

### PR DESCRIPTION
**Pull request by schmidt-sebastian**
Please see Android for any datatype changes.

### CODE REVIEW

File: Firestore/core/test/unit/bundle/bundle_serializer_test.cc [hunk 1]
```diff
@@ -598,29 +599,6 @@ TEST_F(BundleSerializerTest, DecodesReferenceValues) {
   VerifyFieldValueRoundtrip(value);
 }
 
-TEST_F(BundleSerializerTest, DecodeReferenceValuesFromOtherProjectsFails) {
```
Comment (schmidt-sebastian) [inline Firestore/core/test/unit/bundle/bundle_serializer_test.cc:601]: This now logs a debug message when the reference is accessed.

File: Firestore/core/test/unit/bundle/bundle_serializer_test.cc [hunk 2]
```diff
@@ -598,29 +599,6 @@ TEST_F(BundleSerializerTest, DecodesReferenceValues) {
...
-
-  std::string json_string;
-  MessageToJsonString(document, &json_string);
-
-  VerifyJsonStringDecodeFails(json_string);
-}
-
-TEST_F(BundleSerializerTest, DecodeInvalidReferenceFails) {
-  ProtoValue value;
-  value.set_reference_value("projectxx/p1/datmm/new/documents/bundle/test_doc");
```
Comment (schmidt-sebastian) [inline Firestore/core/test/unit/bundle/bundle_serializer_test.cc:615]: This now causes an assert in the client...

Comment (wu-hui) [reply Firestore/core/test/unit/bundle/bundle_serializer_test.cc:615]: But does this mean a corrupted bundle can crash the SDK now?

Comment (schmidt-sebastian) [reply Firestore/core/test/unit/bundle/bundle_serializer_test.cc:615]: If the path is not syntactically valid, the yes. If the path is valid but points to a different project, then we log a warning. It's the same on Android (https://github.com/firebase/firebase-android-sdk/blob/7be699f74d6effca46e765d22bd7635d78d17253/firebase-firestore/src/main/java/com/google/firebase/firestore/bundle/BundleSerializer.java#L267) and on Web, where we just pass the entire message through. The alternative is to pre-parse and validate all data, but that defeats the purpose of the FieldValue to Protobuf migration a bit.

**Review by wu-hui — APPROVED** [review]
LGTM, thank you!

**Pull request merged**